### PR TITLE
ROX-10206: Use modified alerts option map on Postgres to fix autocomplete

### PR DESCRIPTION
**Pull request by connorgorman**
## Description

This fixes _some_ autocomplete on alerts. I have another PR that fixes it more generically for different types. But autocomplete on alerts were failing due to field path differences between alert and list alert. This resolves them by using the alert option path

I filed [ROX-12673](https://issues.redhat.com/browse/ROX-12673) for Post-MVP

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

CI covers lots of alert searching

### CODE REVIEW

File: proto/storage/alert.proto [hunk 1]
```diff
@@ -126,7 +126,7 @@ message Alert {
     repeated Violation        violations        = 5  [(gogoproto.moretags) = 'sensorhash:"ignore" search:"-"'];
     ProcessViolation          process_violation = 13 [(gogoproto.moretags) = 'search:"-"'];
 
-    Enforcement               enforcement       = 6 [(gogoproto.moretags) = 'search:"Enforcement"'];
+    Enforcement               enforcement       = 6;
```
Comment (connorgorman) [inline proto/storage/alert.proto:129]: Enforcement is a message so this is actually unsupported and caused the search walker to crash

**Review by rhybrillou — APPROVED** [review]

File: central/alert/mappings/options.go [hunk 2]
```diff
@@ -3,10 +3,28 @@ package mappings
 import (
 	v1 "github.com/stackrox/rox/generated/api/v1"
 	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/features"
 	"github.com/stackrox/rox/pkg/search"
 )
 
 // TODO: [ROX-10206] Reconcile storage.ListAlert search terms with storage.Alert
```
Comment (dashrews78) [inline central/alert/mappings/options.go:10]: nit:  remove?

**Review by dashrews78 — APPROVED** [review]

File: central/alert/mappings/options.go (hunk 2 above)
Comment (connorgorman) [inline central/alert/mappings/options.go:10]:
```suggestion

```

**Pull request merged**